mat64/doc.go

```diff
@@ -2,38 +2,119 @@
-// into a concrete type of the corresponding value. If there are specific special
-// cases that are needed, please submit a pull-request or file an issue.
+// Package mat64 provides implementations of float64 matrix structures and
+// linear algebra operations on them.
+//
+// Overiew
+//
+// This section provides a quick overview of the matrix package. The following
+// sections provide more in depth commentary.
+//
+// mat64 provides:
+//  - Interfaces for Matrix classes (Matrix, Symmetric, Triangular)
+//  - Concrete implementations (Dense, SymDense, TriDense)
+//  - Methods and functions for using matrix data (Add, Trace, SymRankOne)
+//  - Types for constructing and using matrix factorizations (QR, LU)
+//
+// A matrix may be constructed through the corresponding New function. If no
+// backing array is provided the matrix will be initialized to all zeros.
+//  // Allocate a zeroed array of size 3×5
+//  zero := mat64.NewDense(3, 5, nil)
+// If a backing data slice is provided, the matrix will have those elements.
+// Matrices are all stored in row-major format.
+//  // Generate a 6×6 matrix of random values.
+//  data := make([]float64, 36)
+//  for i := range data {
+//		data[i] = rand.NormFloat64()
+//  }
+//  a := mat64.NewDense(6,6, data)
+//
+// Operations involving matrix data are implemented as functions when the values
+// of the matrix remain unchanged
+//  tr := mat64.Trace(a)
+// and are implemented as methods when the operation modifies the receiver.
+//  zero.Copy(a)
+//
+// Receivers must be the correct size for the matrix operations, otherwise the
+// operation will panic. As a special case for convenience, a zero-sized matrix
+// will be modified to have the correct size, allocating data if necessary.
+//  var c mat64.Dense // construct a new zero-sized matrix
+//  c.Mul(a, a)       // c is automatically adjusted to be 6×6
+//
+// The Matrix Interface(s)
+//
+// The Matrix interface is the common link between the concrete types. The Matrix
+// interface is defined by three functions: Dims, which returns the dimensions
+// of the Matrix, At, which returns the element in the specified location, and
+// T, which will be discussed later. All of the concrete types can perform these
+// behaviors and so implement the interface. Methods and functions are designed
+// to use this interface, so in particular the method
+//  func (m *Dense) Mul(a, b Matrix)
+// constructs a *Dense from the result of a muliplication with any Matrix types,
+// not just *Dense. Where more restrictive requirements must be met, there are also the
+// Symmetric and Triangular interfaces. For example, in
+//  func (s *SymDense) AddSym(a, b Symmetric)
+// the Symmetric interface guarantees a symmetric result.
+//
+// Transposes
+//
+// The T method is used for transpostion. For example, c.Mul(a.T(), b) computes
+// c = a^T * b. The mat64 types implement this method using an implicit transpose.
```

Comment:
Is "See the Transpose type for more information" a good answer? I think this is one of those cases where the source code is easier to understand than the english.